libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,452 @@
+  PAGE_UP,
+  RIGHT_ARROW,
+  UP_ARROW,
+} from '@angular/cdk/keycodes';
+import { Platform } from '@angular/cdk/platform';
+import {
+  AfterViewInit,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  clamp,
+  isDefined,
+  roundToDecimal,
+} from '@dynatrace/barista-components/core';
+import { DtInput } from '@dynatrace/barista-components/input';
+import {
+  animationFrameScheduler,
+  BehaviorSubject,
+  fromEvent,
+  merge,
+  Observable,
+  of,
+  Subject,
+} from 'rxjs';
+import {
+  debounceTime,
+  distinctUntilChanged,
+  filter,
+  map,
+  observeOn,
+  share,
+  switchMap,
+  takeUntil,
+  withLatestFrom,
+} from 'rxjs/operators';
+import {
+  getKeyCodeValue,
+  getSliderPositionBasedOnValue,
+  getSliderValueForCoordinate,
+  roundToSnap,
+} from './slider-utils';
+
+/** We need unique ids in order to have correct labeling. */
+let uniqueId = 0;
+
+const KEY_CODES_ARRAY: Array<number> = [
+  LEFT_ARROW,
+  DOWN_ARROW,
+  RIGHT_ARROW,
+  UP_ARROW,
+  HOME,
+  END,
+  PAGE_UP,
+  PAGE_DOWN,
+];
+
+declare const window: any;
+
+@Component({
+  selector: 'dt-slider',
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.scss'],
+  host: {
+    class: 'dt-slider',
+  },
+  encapsulation: ViewEncapsulation.Emulated,
+  exportAs: 'dtSlider',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DtSlider implements AfterViewInit, OnDestroy, OnInit {
+  /**
+   * Holds the value, the rounding is shifted with.
+   * Calculated based in the step, to avoid JS rounding problems.
+   */
+  private _roundShift: number = 0;
+
+  /** @internal Unique id for this input. */
+  _labelUid = `dt-slider-label-${uniqueId++}`;
+
+  /** Holds the value of the slider. */
+  private _value$ = new BehaviorSubject<number>(0);
+  /** Holds the description of the size of the slider. */
+  private _clientRect$: Observable<ClientRect>;
+  /** Observer that gets triggered if the slider is resized on the screen. */
+  private _resizeObserver$ = new Subject<void>();
+  /** Variable to hold the ResizeObserver */
+  private _observer: any;
+  /** Observer that gets triggered if the input field value is changed. */
+  private _inputFieldValue$ = new Subject<number>();
+
+  /**
+   * Binding for the minimum value of the slider.
+   */
+  @Input()
+  get min(): number {
+    return this._min;
+  }
+  set min(min: number) {
+    this._min = coerceNumberProperty(min);
+    if (this._min > this._value) {
+      this._updateValue(this._min, true);
+    } else {
+      this._updateSliderPosition(this._value, this._min, this._max);
+    }
+  }
+  private _min: number = 0;
+
+  /**
+   * Binding for the maximum value of the slider.
+   */
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+  set max(max: number) {
+    this._max = coerceNumberProperty(max);
```

Comment:
Add a `isDefined` check before you coerce the number because the coercion of `undefined` or `null` will result in `0` and you don't want that. So please handle the case if someone is passing `undefined` or `null`. (Maybe reset to the defaults)

Reply:
I'll add the check, thanks.